articles/quickstart/native/ionic-react/01-login.md

```diff
@@ -0,0 +1,262 @@
+
+Once that's complete, verify that Auth0 redirects you back to your application.
+:::
+
+## Add Logout to Your Application
+
+Now that you can log in, you need [a way to log out](https://auth0.com/docs/logout/guides/logout-auth0). To do this, create a logout button using the `logout` function from the `useAuth0()` hook. Executing `logout` redirects your users to your [Auth0 logout endpoint](https://auth0.com/docs/api/authentication?javascript#logout) (`https://YOUR_DOMAIN/v2/logout`) and then immediately redirects them to your application.
+
+Similar to `buildAuthorizeUrl`, there is an equivalent method `buildLogoutUrl` that can be opened using Capacitor's Browser plugin. However, you will also need to pass a parameter to `logout` that will clear the SDK's state but not perform the redirect to the logout endpoint, as you want to instead do the redirect with the Browser plugin.
+
+Create a new file `LogoutButton.tsx` and add the following code to the file. Then, add the `LogoutButton` component to your app.
+
+```js
+import { useAuth0 } from "@auth0/auth0-react";
+import { Browser } from "@capacitor/browser";
+import { IonButton } from "@ionic/react";
+
+// This should reflect the URL added earlier to your "Allowed Logout URLs" setting
+// in the Auth0 dashboard.
+const callbackUri = "YOUR_PACKAGE_ID://${account.namespace}/capacitor/YOUR_PACKAGE_ID/callback";
+
+const LogoutButton: React.FC = () => {
+  const { buildLogoutUrl, logout } = useAuth0();
+
+  const doLogout = async () => {
+    // Open the browser to perform a logout
+    await Browser.open({ url: buildLogoutUrl({ returnTo: callbackUri }) });
+
+    // Ask the SDK to log out locally, but not do the redirect
+    logout({ localOnly: true });
+  };
+
+  return <IonButton onClick={doLogout}>Log out</IonButton>;
+};
+
+export default LogoutButton;
+```
+
+:::panel Checkpoint
+Add the `LogoutButton` component to your application. When you click it, verify that your Ionic application redirects you the address you specified as one of the "Allowed Logout URLs" in the "Settings" and that you are no longer logged in to your application.
+:::
+
+## Show User Profile Information
+
+The Auth0 React SDK helps you retrieve the [profile information](https://auth0.com/docs/users/concepts/overview-user-profile) associated with logged-in users quickly in whatever component you need, such as their name or profile picture, to personalize the user interface. The profile information is available through the `user` property exposed by the `useAuth0()` hook. Take this `Profile` component as an example of how to use it:
+
+```js
+import { useAuth0 } from "@auth0/auth0-react";
+
+const Profile: React.FC = () => {
+  const { user, isAuthenticated, isLoading } = useAuth0();
+
+  // If the SDK is not ready, or a user is not authenticated, exit.
+  if (isLoading || !isAuthenticated) return null;
+
+  return (
+    <div>
+      <img src={user.picture} alt={user.name} />
```

Comment:
`user?.picture` I got a ts error in your sample app because user can be `undefined` (annoyingly it can't infer this from `isAuthenticated`)

Reply:
Thanks for the call around this. I've updated this code snippet + the sample.